web/app.js

```diff
@@ -728,7 +736,10 @@ let PDFViewerApplication = {
     let url = this.baseUrl;
     // Use this.url instead of this.baseUrl to perform filename detection based
     // on the reference fragment as ultimate fallback if needed.
-    let filename = getPDFFileNameFromURL(this.url);
+    let filename = this.attachedFileName;
+    if (filename === null) {
+      filename = getPDFFileNameFromURL(this.url);
+    }
```

Comment:
Similar to https://github.com/mozilla/pdf.js/pull/8688/files#r128925411, this can be `let filename = this.attachedFileName || getPDFFileNameFromURL(this.url);` instead.